### PR TITLE
refactor(event): update EventCardViewModel for persistent likes

### LINKED ISSUE #408: Update EventCardViewModel and screens to use persistent favorites
### Description
Refactor the EventCardViewModel to use the new repository methods and update UI screens to display persistent likes.

### Acceptance Criteria
- [ ] Rewrite EventCardViewModel to use UserRepository
- [ ] Implement optimistic updates and error handling
- [ ] Update FeedScreen, MapScreen, and EventDetailScreen
- [ ] Add comprehensive ViewModel tests
- [ ] Remove singleton pattern usage

### PR DESCRIPTION
**Pull request by M-e-d-A**
# Description:
## Purpose of the change
This PR implements the front-end logic for the like button using the persistent favorites layer, as defined in sub-issue #408.

It includes:
* Refactored `EventCardViewModel` to use `UserRepository` for persistence
* Optimistic updates and error handling for like operations
* Unit tests for the new `EventCardViewModel` logic
* Updated `EventDetailScreen`, `FeedScreen`, and `MapScreen` to use the new `EventCardViewModel`

## Related issues
Closes #408
Part of #405

## How to test
1. Check out the branch.
2. Run the tests in `EventCardViewModelTest.kt` via Android Studio or `./gradlew test`.
3. Verify all tests pass.
4. Build and run the app, and test liking events on the feed, map, and event detail screens.
5. Verify likes persist after app restart.

**Note:** 
- An LLM was used for the sole and exclusive purpose of improving the clarity and formatting of the PR description.
- This is PR 2 of 4 in the Like & Recommender implementation series.

**Related PRs in this series:**
- #411
- #412 *(this PR)*
- #413
- #414

This PR completes the like functionality by connecting the persistence layer from PR 1 with the UI, providing users with a seamless liking experience across all event screens.

### CODE REVIEW

File: app/src/main/java/ch/onepass/onepass/ui/map/MapScreen.kt [hunk 1]
```diff
@@ -63,10 +63,10 @@ fun MapScreen(
     mapViewModel: MapViewModel = viewModel(),
     filterViewModel: EventFilterViewModel = viewModel(),
     onNavigateToEvent: (String) -> Unit = {},
+    eventCardViewModel: EventCardViewModel = viewModel()
```
Comment (Hassan372) [inline app/src/main/java/ch/onepass/onepass/ui/map/MapScreen.kt]: You've got `eventCardViewModel` as both a parameter and with a default value. Pick one approach.

**Before:**
```kotlin
eventCardViewModel: EventCardViewModel = viewModel()
```

**After (Option A):**
```kotlin
eventCardViewModel: EventCardViewModel
```

**After (Option B):**
```kotlin
// Remove parameter, create locally
val eventCardViewModel: EventCardViewModel = viewModel()
```

I'd go with Option A if you want to control the ViewModel from the parent (easier to test). Option B if you want each screen to own its own instance.

File: app/src/main/java/ch/onepass/onepass/ui/event/EventCardViewModel.kt [hunk 2]
```diff
@@ -1,27 +1,91 @@
...
+    viewModelScope.launch {
+      val result =
+          if (isLiked) {
+            userRepository.removeFavoriteEvent(uid, eventId)
+          } else {
+            userRepository.addFavoriteEvent(uid, eventId)
+          }
+
+      // Revert if failed
+      if (result.isFailure) {
```
Comment (Hassan372) [inline app/src/main/java/ch/onepass/onepass/ui/event/EventCardViewModel.kt]: Right now when persistence fails, it just silently rolls back. Adding a log would help you debug issues later.

**Before:**
```kotlin
if (result.isFailure) {
    _likedEvents.update { if (isLiked) it - eventId else it + eventId }
}
```

**After:**
```kotlin
if (result.isFailure) {
    Log.e("EventCardViewModel", "Failed to persist like for event $eventId", result.exceptionOrNull())
    _likedEvents.update { if (isLiked) it - eventId else it + eventId }
}
```

**Review by Hassan372 — CHANGES_REQUESTED** [review]
Nice refactoring here. You've taken the singleton pattern and replaced it with proper dependency injection, which is definitely the right move. The test coverage looks solid - you're handling edge cases like failures and null users. I particularly like the optimistic updates with rollback on error, that'll make the UI feel snappy.

The PR description is clear and easy to follow. Good use of `StandardTestDispatcher` in the tests too.

Please take a look at the comments I made

File: app/src/main/java/ch/onepass/onepass/ui/map/MapScreen.kt (hunk 1 above)
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/map/MapScreen.kt]: Thanks. We have standardized on Option A: Parameter Injection with Default Value (as implemented for EventDetailScreen and FeedScreen).

File: app/src/main/java/ch/onepass/onepass/ui/eventdetail/EventDetailScreen.kt [hunk 3]
```diff
@@ -66,14 +66,14 @@ fun EventDetailScreen(
...
-            factory = viewModelFactory { initializer { EventDetailViewModel(eventId = eventId) } })
+            factory = viewModelFactory { initializer { EventDetailViewModel(eventId = eventId) } }),
+    eventCardViewModel: EventCardViewModel = viewModel()
 ) {
   val event by viewModel.event.collectAsState()
   val organization by viewModel.organization.collectAsState()
   val isLoading by viewModel.isLoading.collectAsState()
   val error by viewModel.error.collectAsState()
 
-  val eventCardViewModel = EventCardViewModel.getInstance()
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/eventdetail/EventDetailScreen.kt]: Agreed. The getInstance() pattern is being replaced with the standard Jetpack Compose/Hilt pattern of Parameter Injection with Default Value (viewModel()). This ensures the component is testable while retaining simple usage in production code (like the AppNavHost).

File: app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt [hunk 4]
```diff
@@ -65,6 +65,7 @@ fun FeedScreen(
     onNavigateToNotifications: () -> Unit = {},
     viewModel: FeedViewModel = viewModel(),
     filterViewModel: EventFilterViewModel = viewModel(),
+    eventCardViewModel: EventCardViewModel = viewModel()
```
[the remark this answers is not in the document]
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/feed/FeedScreen.kt]: Agreed. EventCardViewModel is added as an injectable parameter to FeedScreen and passed down to the EventListContent (which renders the cards).

File: app/src/main/java/ch/onepass/onepass/ui/event/EventCardViewModel.kt (hunk 2 above)
Comment (M-e-d-A) [reply app/src/main/java/ch/onepass/onepass/ui/event/EventCardViewModel.kt]: Agreed. We have added android.util.Log.e to capture and log the exception when the repository call (addFavoriteEvent or removeFavoriteEvent) fails, making the failure condition visible in Logcat for debugging. The state rollback logic remains, correctly handling the optimistic update failure.
